### ISSUE TITLE
Create a proper CI/CD

### ISSUE THREAD

**Issue by XenorPLxx** [sharenite/sharenite#20]
This will probably have to be done after #7, since #7 will include significant changes to the docker images, and CI/CD should build out those images, test against them (RSpec, Rubocop, Lint), push to an image repository and then trigger a deployment.

Probably should be also connected to #14 and #19

**Comment by XenorPLxx** [comment]
Triage update (2026-03-02):
- Status: `not covered`
- Priority: `P1`
- Makes sense: `yes`
- Notes: Still important. Recent safe deploy scripts reduce risk, but full CI/CD remains open.

**Comment by XenorPLxx** [comment]
Closing as intentionally out of scope for now: we are using local/manual deploy commands instead of setting up CI/CD at this stage.
